File: pi/www/js/audiomanager.js
let sounds = {};

class AudioManager{

    init(fileObjects){
        for(let i = 0; i < fileObjects.length; i++){
            let name = fileObjects[i].filename.split(".")[0];
            this.sound = document.createElement("audio");
            this.sound.src = "/sound/" + fileObjects[i].filename;
            this.sound.setAttribute("preload", "auto");
            this.sound.setAttribute("controls", "none");
            this.sound.style.display = "none";
            this.sound.volume = fileObjects[i].volume;
            this.sound.loop = fileObjects[i].loop;
            document.body.appendChild(this.sound);
            fileObjects[i]["sound"] = this.sound;
            sounds[name] = fileObjects[i];
        }
    }

    play(name){
        this.stop(name);
        sounds[name].sound.play();
    }

    stop(name){
        sounds[name].sound.pause();
        sounds[name].sound.currentTime = 0;
    }
}

File: app/android/Microwaffle400/app/src/main/java/com/soloheisbeer/microwaffle400/audio/AudioManager.kt
package com.soloheisbeer.microwaffle400.audio

import android.content.Context
import android.media.MediaPlayer

class AudioManager (val context: Context) {

    var sounds = mutableMapOf<String, MediaPlayer>()

    fun init() {
        val fields = arrayListOf(
            "alarm",
            "boot",
            "down",
            "loop",
            "up"
        )
        for (field in fields) {
            val mp = MediaPlayer.create(context, context.resources.getIdentifier(field, "raw", context.packageName))
            sounds[field] = mp
        }
    }

    fun play(name: String, loop: Boolean = false, volume: Float = 1.0f){
        val sound = sounds[name] ?: return
        sound.isLooping = loop
        sound.setVolume(volume, volume)

        stop(name)
        sound.start()
    }

    fun stop(name: String){
        val sound = sounds[name] ?: return
            sound.stop()
            sound.prepare()
    }

    fun cleanUp() {
        for(mp in sounds.values){
            mp.reset()
            mp.release()
        }
    }
}